Shared/API/CloudFunction.swift

```diff
@@ -11,11 +11,66 @@ import Parse
 import Combine
 
 protocol CloudFunction: StatusableRequest {
-    func makeRequest(andUpdate statusables: [Statusable], viewsToIgnore: [UIView]) -> AnyPublisher<ReturnType, Error>
+    func makeRequest(andUpdate statusables: [Statusable],
+                     viewsToIgnore: [UIView]) -> AnyPublisher<ReturnType, Error>
+    func makeAsyncRequest(andUpdate statusables: [Statusable],
+                          viewsToIgnore: [UIView]) async throws -> ReturnType
 }
 
 extension CloudFunction {
 
+    func makeAsyncRequest(andUpdate statusables: [Statusable],
+                          params: [String : Any],
+                          callName: String,
+                          delayInterval: TimeInterval = 2.0,
+                          viewsToIgnore: [UIView]) async throws -> Any {
+
+        // Trigger the loading event for all statusables
+        for statusable in statusables {
+            statusable.handleEvent(status: .loading)
+        }
+
+        // Reference the statusables weakly in case they are deallocated before the signal finishes.
+        let weakStatusables: [WeakAnyStatusable] = statusables.map { (statusable)  in
+            return WeakAnyStatusable(statusable)
+        }
+
+        return try await withCheckedThrowingContinuation({ (continuation: CheckedContinuation<Any, Error>) in
+
+            PFCloud.callFunction(inBackground: callName,
+                                 withParameters: params) { (object, error) in
+
+                if let error = error {
+                    SessionManager.shared.handleParse(error: error)
+                    weakStatusables.forEach { (statusable) in
+                        statusable.value?.handleEvent(status: .error(error.localizedDescription))
+                    }
+
+                    continuation.resume(throwing: error)
+                } else if let value = object {
+                    weakStatusables.forEach { (statusable) in
+                        statusable.value?.handleEvent(status: .saved)
+                    }
+
+                    continuation.resume(returning: value)
```

Comment:
We say we're done by calling "resume". This must only be called once. Thankfully the compiler can tell if we accidentally call it twice.